src/relay/backend/contrib/cmsisnn/tir_to_runtime.cc

```diff
@@ -478,10 +466,28 @@ class CodeGenCMSISNN : public codegen::CodeGenCHost {
     stream << "&" << input_dim << ", " << input_data << ", ";
     stream << "&" << filter_dim << ", ";
     stream << "&" << output_dim << ", " << output_data << ");\n";
+    EmitErrorCheck();
+  }
+
+  void EmitErrorCheck() {
+    auto emit_error = [&](std::string error) {
+      stream << "TVMAPISetLastError(\"" << error << "\");";
+    };
+
     PrintIndent();
-    stream << "if (status != ARM_CMSIS_NN_SUCCESS) {\n";
+    stream << "switch (status) {\n";
+    PrintIndent();
+    PrintIndent();
+    stream << "case ARM_CMSIS_NN_SUCCESS: break;\n";
+    PrintIndent();
+    PrintIndent();
```

Comment:
Potential fun exercise for the future, add a parameter to `PrintIndent` for the number of indents 😸